src/coreclr/src/vm/diagnosticsprotocol.h

```diff
@@ -516,8 +530,14 @@ namespace DiagnosticsIpc
             }
             CONTRACTL_END;
 
-            const U* payload = reinterpret_cast<const U*>(m_pData);
-            m_pData = nullptr; // user is expected to clean up buffer when finished with it
+            // don't have std::is_trivially_copyable
+            // static_assert(std::is_trivially_copyable<U>::value, "Can only directly parse trivially copyable types.");
+
+            const U *payload = new (nothrow) U;
+            if (payload == nullptr)
+                return nullptr;
+            memcpy((void *)payload, m_pData, sizeof(U));
+            delete[] m_pData;
```

Comment:
Can `delete[]` be used when paired with a non-array `new`?

Reply:
`m_pData` is `new`'d up above using `new (nothrow) BYTE[m_Size]`

Reply:
Oh, I misread. Thought it was `payload`.